### ISSUE TITLE
Access to generated private types

### ISSUE THREAD

**Issue by noobloser** [rrousselGit/freezed#337]
**Is your feature request related to a problem? Please describe.**
I intend to implement the library [reflectable](https://pub.dev/packages/reflectable) with some of my freezed types. Here I want to use the reflector.reflectType() method to get runtime access to my freezed class members.
The issue is, that the abstract class is evaluated as a Type and not a class (which seems to be right imo).
So I somehow would like to gain access to the 'real' class definition of freezed.

**Describe the solution you'd like**
example freezed definition:
```
@freezed
abstract class Tester implements _$Tester {
  const factory Tester(String value) = TesterT;
}
```

in the generated "tester.freezed.dart" file there is a regular private class generated with the name _$TesterT.
It would be awesome, if one could change the freezed definition, so the generated type is generated public, thus usable in other files.
I thought about maybe changing the statement like this, which could generate public types.

```
@freezed
abstract class Tester implements $Tester {
  const factory Tester(String value) = TesterT;
}
```

**Comment by rrousselGit** [comment]
I'm not quite sure. Why would you need access to the `_$Class` when you already have a public one?

In your example, can't you use `Tester` or `TesterT`?

**Comment by noobloser** [comment]
Thanks for the feedback. Because I solved my use case otherwise already, this is just out of curiosity (or for others).
As I understood [reflectable](https://pub.dev/packages/reflectable) I needs a non abstract class definition which neither Tester nor TesterT is. 

By now I have figured out, that my approach to [reflectable](https://pub.dev/packages/reflectable) is not even working with a regular class type. So this is not an issue of freezed at all. I'm sorry for wasting your time.
